projects/solar/src/relay_fsm.c

```diff
@@ -2,17 +2,64 @@
 
 #include <stdbool.h>
 
+#include "can_transmit.h"
+#include "can_unpack.h"
+#include "data_store.h"
 #include "event_queue.h"
 #include "fault_handler.h"
 #include "fsm.h"
+#include "gpio.h"
 #include "log.h"
+#include "solar_boards.h"
 #include "solar_config.h"
 #include "solar_events.h"
 #include "status.h"
 
+static SolarMpptCount solar_mppt_count;
+
+static void prv_assert_relay(SoftTimerId timer_id, void *context) {
+  RelayFsmStorage *storage = (RelayFsmStorage *)context;
+  if (storage->isErrCalled == true) {
+    fault_handler_raise_fault(EE_SOLAR_RELAY_OPEN_ERROR, 0);
+    return;
+  }
+
+  int32_t data_value;
+  bool isSet = false;
+
+  data_store_get_is_set(DATA_POINT_CURRENT, &isSet);
+
+  if (!isSet) {
+    if (storage->isSetCounter >= 30) {
+      LOG_DEBUG("Aborting, The current is not set\n");
+      fault_handler_raise_fault(EE_SOLAR_RELAY_OPEN_ERROR, 0xFF);
+      return;
+    }
+    storage->isSetCounter += 1;
+    soft_timer_start_millis(DATA_STORE_ASSERTION_DELAY_MS, prv_assert_relay, context, NULL);
+    return;
+  }
+
+  data_store_get(DATA_POINT_CURRENT, (uint32_t *)&data_value);
+
+  if (data_value > CURRENT_ASSERT_THRESHOLD_uA) {
+    fault_handler_raise_fault(EE_SOLAR_RELAY_OPEN_ERROR, data_value);
+    return;
+  } else {
+    // Success message
+    if (solar_mppt_count == 6) {
+      CAN_TRANSMIT_RELAY_CURRENT_6_MPPTS();
+    } else if (solar_mppt_count == 5) {
```

Comment:
Use `SOLAR_BOARD_5/6_MPPTS` from solar_boards.h instead of plain 5 and 6, that way e.g. if we add another one it's easier to figure out where to change.

Reply:
done